build/devenv/tests/e2e/smoke_test.go

```diff
@@ -257,54 +257,81 @@ func TestE2ESmoke(t *testing.T) {
 	})
 
 	t.Run("extra args v3 token transfer", func(t *testing.T) {
-		tcs := []struct {
-			name    string
-			src     uint64
-			dest    uint64
-			amount  *big.Int
-			timeout time.Duration
-		}{
+		type tokenTransferTestCase struct {
+			name                    string
+			src                     uint64
+			dest                    uint64
+			amount                  *big.Int
+			tokenQualifier          string
+			expectedReceiptIssuers  int
+			expectedVerifierResults int
+		}
+
+		tcs := []tokenTransferTestCase{
+			{
+				name:                    "burn&mint EOA executes; receiver increases; sender decreases (TEST)",
+				src:                     selectors[0],
+				dest:                    selectors[1],
+				amount:                  big.NewInt(1000),
+				tokenQualifier:          "TEST",
+				expectedReceiptIssuers:  3,
+				expectedVerifierResults: 1,
+			},
 			{
-				name:    "burn&mint EOA executes; receiver increases; sender decreases",
-				src:     selectors[0],
-				dest:    selectors[1],
-				amount:  big.NewInt(1000),
-				timeout: 45 * time.Second,
+				name:                    "burn&mint EOA executes; receiver increases; sender decreases; default is required (NOCCV)",
+				src:                     selectors[0],
+				dest:                    selectors[1],
+				amount:                  big.NewInt(1000),
+				tokenQualifier:          "NOCCV",
+				expectedReceiptIssuers:  3,
+				expectedVerifierResults: 1,
+			},
+			{
+				name:                    "burn&mint EOA executes; receiver increases; sender decreases; default and secondary are required (SECONDARY)",
+				src:                     selectors[0],
+				dest:                    selectors[1],
+				amount:                  big.NewInt(1000),
+				tokenQualifier:          "SECONDARY",
+				expectedReceiptIssuers:  4,
+				expectedVerifierResults: 2,
+			},
+			{
+				name:                    "burn&mint EOA executes; receiver increases; sender decreases (DUAL)",
+				src:                     selectors[0],
+				dest:                    selectors[1],
+				amount:                  big.NewInt(1000),
+				tokenQualifier:          "DUAL",
+				expectedReceiptIssuers:  4,
+				expectedVerifierResults: 2,
 			},
 		}
+
 		for _, tc := range tcs {
 			t.Run(tc.name, func(t *testing.T) {
 				receiver := mustGetEOAReceiverAddress(t, c, tc.dest)
-				// Sender is the chain deployer key used by the env
 				sender := protocol.UnknownAddress(e.BlockChains.EVMChains()[tc.src].DeployerKey.From.Bytes())
```

Comment:
This is pretty EVM specific and didn't notice it the first time around - can we add a method to the `Chains` interface to get the sender address instead?